backend/api/views.py

```diff
@@ -60,59 +60,51 @@ def cancel(self, request, pk=None):
     @action(detail=False, methods=['post'], name='Submit new job')
     def submit(self, request):
         serializer = PrintRequestSerializer(data=request.data)
-        if not serializer.is_valid():
-            return Response(serializer.errors,
-                            status=status.HTTP_400_BAD_REQUEST)
+        serializer.is_valid(raise_exception=True)
         printer_with_perms = Printer.get_printer_for_user(user=self.request.user,
                                                           printer_id=serializer.validated_data['printer'])
         if not printer_with_perms:
-            return Response("Printer does not exist", status=status.HTTP_400_BAD_REQUEST)
+            raise ParseError("Selected printer does not exist")
 
         try:
             job = self._create_printing_job(printer_with_perms=printer_with_perms, **serializer.validated_data)
             self._upload_artefact(job, **serializer.validated_data)
             self._run_job(job)
         except UnsupportedDocumentError as ex:
-            # FIXME: Use a common error message format
-            return Response("Error: {}".format(ex), status=status.HTTP_400_BAD_REQUEST)
+            raise ParseError(str(ex))
         return Response(self.get_serializer(job).data)
 
     @action(detail=True, methods=['post'], name='Upload artefact')
     def upload_artefact(self, request, pk=None):
         job = self.get_object()
         if job.status != JobStatus.INCOMING:
-            return Response("Error: invalid job status for this request", status=status.HTTP_400_BAD_REQUEST)
+            raise ParseError(["Invalid job status for this request"," current status: {}".format(job.status)])
```

Comment:
Why pass the details as a list here?

Reply:
This is also not a `ParseError`, I suggest creating a custom `ApiError` subclass and use the HTTP status code `422 Unprocessable Content`